ggshield/iac/filter.py

```diff
@@ -51,6 +50,8 @@ def get_iac_files_from_paths(
 def is_file_iac_file(scannable: Scannable) -> bool:
     if any(ext in IAC_EXTENSIONS for ext in scannable.path.suffixes):
         return True
+    if any(scannable.path.name.endswith(iac_ext) for iac_ext in IAC_EXTENSIONS):
+        return True
```

Comment:
Are you keeping both on purpose?

If not:

```suggestion
def is_file_iac_file(scannable: Scannable) -> bool:
    if any(scannable.path.name.endswith(iac_ext) for iac_ext in IAC_EXTENSIONS):
        return True
```

Reply:
Yes, they're different.
Before, the `.py.schema` files returned True only because the `.py` extension existed.
Now:
If `.tf` is in any extension of the file (like `something.tf.other`) we return True.
If `.py.schema` is the ending pattern of the file, we return True as well.
In the end, I reckon we will overhaul that part since it does particularly efficient nor well-designed.
But for now, it handles the patterns correctly.